packages/calypso-data-stores/package.json

```diff
@@ -0,0 +1,51 @@
+{
+	"name": "@automattic/calypso-data-stores",
+	"version": "1.0.0-alpha.0",
+	"description": "Calypso Data Stores",
+	"homepage": "https://github.com/Automattic/wp-calypso",
+	"license": "GPL-2.0-or-later",
+	"author": "Automattic Inc.",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
+	"sideEffects": [
+		"*.css",
+		"*.scss"
+	],
```

Comment:
I don't believe we have these here 🙂 however we'll likely need a good `sideEffects` declaration. It's worth spending a bit of time getting that right or there's potential to pull in excess code and mount redundant stores.

Reply:
Good point about styles :sweat_smile: Copy-pasta :grimacing: 

And agree about the side effects. I guess I'll try separating store registration from definitions. (I wish there was a side-effect free `createStore` whose result could be fed into `registerSore`, but I couldn't find anything to that effect.)

Reply:
Re side-effects: your modules register the stores statically during the module initialization. When working on the Gutenlypso integration a long time ago, this practice gave us some unpleasant headaches. When code-splitting modules and async-loading them, you have very limited control over when exactly is a store registered. The `@wordpress/data` registry must be already initialized at the time when a store module is loaded.

Also, calling `registerStore` in your module assumes that there is ever only one registry. That's currently true with `@wordpress/data` -- it has one singleton registry inside it and it's impossible to have more instances -- but that's a design flaw that will probably be fixed in the future. (Think of using `@wordpress/data` for SSR -- it's impossible).

I'd recommend exporting the thing that's passed to `registerStore` (it's called "store config" I think) from the module and do the registration in app code.

Reply:
> Re side-effects: your modules register the stores statically during the module initialization. When working on the Gutenlypso integration a long time ago, this practice gave us some unpleasant headaches. When code-splitting modules and async-loading them, you have very limited control over when exactly is a store registered. The `@wordpress/data` registry must be already initialized at the time when a store module is loaded.
> 
> Also, calling `registerStore` in your module assumes that there is ever only one registry. That's currently true with `@wordpress/data` -- it has one singleton registry inside it and it's impossible to have more instances -- but that's a design flaw that will probably be fixed in the future. (Think of using `@wordpress/data` for SSR -- it's impossible).
> 
> I'd recommend exporting the thing that's passed to `registerStore` (it's called "store config" I think) from the module and do the registration in app code.

Thanks @jsnajdr! Yeah, I agree with your points -- it's a bit unfortunate that `registerStore` is so tightly coupled to an implicit, global registry.

I'll go ahead and export the store configs so we'll become side-effects free. Hoping to find a good format for associating the store keys with the configs so I'll be able to provide a nice enough API :thinking: